src/etcd/client.py

```diff
@@ -757,7 +757,8 @@ def eternal_watch(self, key, index=None, recursive=None):
         local_index = index
         while True:
             response = self.watch(key, index=local_index, timeout=0, recursive=recursive)
-            local_index = response.modifiedIndex + 1
+            if index:
```

Comment:
Shouldn't this be ``local_index`` since subsequent runs through the loop will be based off of that?